Scripts/cours.js

```diff
@@ -56,47 +67,74 @@ function deleteAll(displayOnly = true){
 
     for(let i=size-1; i>1; i--){
         class_tab.deleteRow(i);
-    }
+    } 
+    showButton();
 }
 
 function deleteClass(id){
+    console.log(id);
+    if(classes == 0)
+        classes = rawTab;
  const idClass = classes.findIndex(c => c.id === id);
 
  if(idClass !== -1){
     classes.splice(idClass, 1);
     deleteAll();
-    getClasses();
+    displayClasses();
  }else{
     alert("Not found");
  }
 }
 
-function postClass(){
+function createClass(){
     const title = document.getElementById("titleInput").value;
     const period = document.getElementById("periodInput").value;
     const nbHour = document.getElementById("nbHourInput").value;
     const teacher = document.getElementById("teacherInput").value;
     const description = document.getElementById("descriptionInput").value;
 
-    if(!title || !period || !teacher || !description || !Number.isInteger(+nbHour)){
+    // Ctrl input
+    // Number.isInteger(+nbHour) => le + permet de retirer les ""
+    if(!title || !period || !teacher || !description || !nbHour)
+    {
         alert("Vous n'avez pas remplis tous les champs");
-        return;
+        return; 
     }
+    
     const addClass = {
         "id": classes.length+1,
         "label": title,
         "period":period,
-        "nbHour": nbHour,
+        "nbHour":{ 
+            "valeur":nbHour,
+            "unity": "heures"
+            },
         "teacher": teacher,
         "detail": description
     };
     classes.push(addClass);
-    document.getElementById("titleInput").value = "";
-    document.getElementById("periodInput").value = "";
-    document.getElementById("nbHourInput").value = "";
-    document.getElementById("teacherInput").value = "";
-    document.getElementById("descriptionInput").value = "";
+    clearInput(["titleInput","periodInput","nbHourInput", "teacherInput","descriptionInput"]);
     deleteAll();
-    getClasses();
+    displayClasses();
+}
+
+function clearInput(list){
+    for(let i= 0; i< list.length-1; i++){
+         document.getElementById(list[i]).value = "";
+    }
+}
+
+function showButton(){
+    document.getElementById("initializeBtn").style.display = "initial"; 
+}
+
+function initialize(){
+    document.getElementById("initializeBtn").style.display = "none"; 
 
-}
\ No newline at end of file
+    if(classes == 0){
+        // Qd je vide le tableau elmt par elemt je rentre ici
```

Comment:
it's because you call deleteAll in your delete and also because the classes array and rawTab are the same